libs/vkd3d/memory.c

```diff
@@ -1038,6 +1038,7 @@ static HRESULT vkd3d_memory_allocator_flush_clears_locked(struct vkd3d_memory_al
     uint32_t queue_mask, queue_index;
     VkCommandBuffer vk_cmd_buffer;
     VkSubmitInfo submit_info;
+    VkBufferMemoryBarrier *buffer_memory_barriers;
```

Comment:
Don't use `VkBufferMemoryBarrier`, just use a single `VkMemoryBarrier`. Avoids both the memory allocation and the loop, there is no good reason to ever use `VkBufferMemoryBarrier` other than resource ownership transfers.